File: Partnership-Managment/src/main/java/com/partnershipmanagement/Entities/Partnership.java
package com.partnershipmanagement.Entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Partnership {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idPartnership;

    @Enumerated(EnumType.STRING)
    private PartnershipStatus partnershipStatus;

    private float score;

    @ManyToOne

    Entreprise entreprise;

    // Many-to-one relationship: Many applications can be related to one proposal
    @ManyToOne
    @JoinColumn(name = "proposal_id",referencedColumnName = "idProposal")
    @JsonIgnoreProperties("applications")
    private Proposal proposals;


    @OneToMany(cascade = CascadeType.ALL, mappedBy="partnership")
    @ToString.Exclude
    private Set<Assessment> Assesements;




    @Transient
    private Boolean isCreditEligible;

    @Transient
    private Double creditEligibilityProbability;
}

File: Front_end/src/app/PartnershipManagement/models/partnership.ts
import { Assessment } from "./assessment.model";
import { Entreprise } from "./entreprise";
import { Proposal } from "./proposal";

export interface Partnership {
    idPartnership?: number;
    partnershipStatus: string;
    score?: number;

    entreprise:Entreprise;  // Just the ID here, not the full object
    proposals?: Proposal;  // Replace with proper Proposal model if available
    Assesements?: Assessment[]; // Replace with proper Assessment model if available
  }
  